todo/src/App.js

```diff
@@ -1,28 +1,55 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
 import './App.css';
+import TodoForm from './components/TodoForm';
+import TodoList from './components/TodoList';
+import { connect } from 'react-redux';
+import { addTodo, toggleTodo, clearCompleted } from './actions';
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      task: ''
+    }
+  }
+
+  handleChanges = e => {
+    console.log(this.state)
+    this.setState({
+      [e.target.name]: e.target.value
+    })
+  }
+
+  addTodo = e => {
+    e.preventDefault();
+    this.props.addTodo(this.state.task);
+    this.setState({ task: '' })
+  }
+
+  toggleTodo = id => {
+    this.props.toggleTodo(id);
+  }
+
+  clearCompleted = () => {
+    this.props.clearCompleted();
+  }
+
   render() {
     return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <p>
-            Edit <code>src/App.js</code> and save to reload.
-          </p>
-          <a
-            className="App-link"
-            href="https://reactjs.org"
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            Learn React
-          </a>
-        </header>
+      <div className="container">
+        <h1>ToDo List</h1>
+        <TodoList todos={this.props.todos} toggleTodo={this.toggleTodo} />
+        <TodoForm handleChanges={this.handleChanges} addTodo={this.addTodo} task={this.state.task} />
+        <button onClick={this.clearCompleted}>Clear Completed</button>
       </div>
     );
   }
 }
 
-export default App;
+const mapStateToProps = (state) => {
+    return {
+        ...state
+    };
+};
+
+export default connect(mapStateToProps, { addTodo, toggleTodo, clearCompleted })(App);
```

Comment:
for both the mapStateToProps and the actioncreators you're bringing in: they don't need to exist within App except to be passed down to the TodoList and TodoForm components. Remember that with Redux you should be thinking about having each component access the individual pieces of the store's state tree-- or action creators used to change that state tree -- with the connect function instead of having to pass things around as props from parent component to child like this. The way you're doing it, there's basically no reason to use Redux at all since you take everything from your store into App, and then pass those pieces of state or actioncreators down as props like it was a normal React app